plan/task/cacheconfig.go

```diff
@@ -0,0 +1,234 @@
+				}
+				exportConfigStr += ",scope=" + scope
+			}
+
+			if urlVal := v.Lookup("url"); urlVal.Exists() {
+				var err error
+				url, err := urlVal.String()
+				if err != nil {
+					return nil, fmt.Errorf("failed to lookup GHA cache export url: %w", err)
+				}
+				exportConfigStr += ",url=" + url
+			}
+
+			// TODO: this should be a secret, not string
+			if tokenVal := v.Lookup("token"); tokenVal.Exists() {
+				var err error
+				token, err := tokenVal.String()
+				if err != nil {
+					return nil, fmt.Errorf("failed to lookup GHA cache export token: %w", err)
+				}
+				exportConfigStr += ",token=" + token
+			}
+
+		case "registry":
+			ref, err := v.Lookup("ref").String()
+			if err != nil {
+				return nil, fmt.Errorf("failed to lookup registry cache export ref: %w", err)
+			}
+			exportConfigStr += ",ref=" + ref
+		default:
+			return nil, fmt.Errorf("unsupported cache export type: %s", typ)
+		}
+
+		mode, err := exportField.Lookup("mode").String()
+		if err != nil {
+			return nil, fmt.Errorf("failed to lookup cache export mode: %w", err)
+		}
+		exportConfigStr += ",mode=" + mode
+
+		// TODO: remove or lower to debug
+		lg.Info().Msgf("export cache config: %s", exportConfigStr)
+
+		exportOpts, err := parseExportCache([]string{exportConfigStr})
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cache export config: %w", err)
+		}
+		pctx.CacheConfig.SetExports(exportOpts)
+	}
+
+	return compiler.NewValue(), nil
+}
+
+func parseExportCache(exports []string) ([]buildkit.CacheOptionsEntry, error) {
+	cacheExports, err := build.ParseExportCache(exports, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	cacheExports = convertRelativePaths(cacheExports)
+
+	for _, ex := range cacheExports {
+		if err := addGithubActionsAttrs(ex); err != nil {
+			return nil, fmt.Errorf("failed to parse cache export options: %w", err)
+		}
+	}
+
+	return cacheExports, nil
+}
+
+func parseImportCache(imports []string) ([]buildkit.CacheOptionsEntry, error) {
+	cacheImports, err := build.ParseImportCache(imports)
+	if err != nil {
+		return nil, err
+	}
+
+	cacheImports = convertRelativePaths(cacheImports)
+
+	for _, im := range cacheImports {
+		if err := addGithubActionsAttrs(im); err != nil {
+			return nil, fmt.Errorf("failed to parse cache import options: %w", err)
+		}
+	}
+
+	return cacheImports, nil
+}
+
+func convertRelativePaths(cacheOptionEntries []buildkit.CacheOptionsEntry) []buildkit.CacheOptionsEntry {
+	pathableAttributes := []string{"src", "dest"}
+
+	for _, option := range cacheOptionEntries {
+		for _, key := range pathableAttributes {
+			if _, ok := option.Attrs[key]; ok {
+				path, err := homedir.Expand(option.Attrs[key])
+				if err != nil {
+					panic(err)
+				}
+				option.Attrs[key] = path
+			}
+		}
+	}
+
+	return cacheOptionEntries
+}
+
+// TODO: not sure if reading env vars from here is the right place... probably deprecate this behavior and use client env explicitly
```

Comment:
This is already done in [`ParseImportCache`](https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/importcache.go#L40) and [`ParseExportCache`](https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/exportcache.go#L43) functions of buildkit with [`loadGitHubEnv`](https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/util.go#L16) so I'm not even sure you need this function.